src/expint.jl

```diff
@@ -0,0 +1,410 @@
+import Polynomials
+using Base.MathConstants
+
+#function polymult(p1::Vector{T}, p2::Vector{T}) where {T}
+#    n, m = length(p1) - 1, length(p2) - 1
+#    c = zeros(T, m + n + 1)
+#    for i in 0:n, j in 0:m
+#        c[i + j + 1] += p1[i + 1] * p2[j + 1]
+#    end
+#    return c
+#end
+
+function E₁_cfpoly_approx(n::Integer, pstart::Polynomials.Polynomial{T}, ::Type{T}=BigInt) where {T<:Real}
+    q = Polynomials.Polynomial(T[1])
+    p = pstart
+    x = Polynomials.Polynomial(T[0,1])
+    for i = n:-1:1
+        p, q = x*p+(1+i)*q, p # from cf = x + (1+i)/cf = x + (1+i)*q/p
+        p, q = p + i*q, p     # from cf = 1 + i/cf = 1 + i*q/p
```

Comment:
Since the only polynomial operations we need are (1) multiplication by scalars, (2) addition, and (3) multiplication by `x`, I would just define a `struct SimplePoly{T}; Vector{T}; end` that supports these three operations.